Allow validation for not Vert.x JSON types

pmlopes opened the pull request:
Signed-off-by: Paulo Lopes <pmlopes@gmail.com>

Motivation:

Fixes #38 

Currently, the validation only allows `JsonObject` and `JsonArray` for non-basic types. Many libraries could benefit from this project if we would allow besides the vert.x native types `java.util.Map` and `java.util.List`.

This PR addresses the issue by handling the internal of the validation using the JDK types instead of the vert.x ones.

slinkydeveloper commented on src/main/java/io/vertx/json/schema/common/ItemsValidatorFactory.java:
```diff
@@ -41,10 +42,13 @@ public ItemsValidator(MutableStateValidator parent) {
         for (int i = 0; i < arr.size(); i++) {
           context.markEvaluatedItem(i);
-          schema.validateSync(context.lowerLevelContext(), arr.getValue(i));
+          schema.validateSync(context.lowerLevelContext(), JsonUtil.wrapJsonValue(arr.get(i)));
```
Do we really need the wrap when going down the structure? If the structure is something like `List<List<Integer>>`, it should work anyway, right?

slinkydeveloper commented on src/test/java/io/vertx/json/schema/JavaUtilTest.java:
```diff
@@ -0,0 +1,94 @@
+  }
+
+  @Test
+  public void schemaAndValidationUsingList(Vertx vertx) {
```
Can you add 2 additional nested cases, one with list nested inside maps and another one with maps nested inside lists?

slinkydeveloper commented:
Before merging this one, I would love to merge #37 first

pmlopes commented:
I've removed all wrapping as it is not needed anymore now that both maps and lists are supported, this should reduce GC a bit further and added an example from https://json-schema.org/learn/getting-started-step-by-step using plain maps and lists as a test to verify it.

slinkydeveloper commented:
Can you go ahead and rebase?

pmlopes commented:
I've remade the PR as #40
